src/toil/jobStores/googleJobStore.py

```diff
@@ -0,0 +1,536 @@
+            if not readAll and self.fileExists(checkID):
+                continue
+            try:
+                readFn(str(indexToTry), callback)
+                read += 1
+                beenRead.add(indexToTry)
+            except NoSuchFileException:
+                # these are just possible indexes, OK if we don't find a file there
+                pass
+        return read
+
+    def _getPrimaryStatsID(self, index, beenRead=False):
+        newID = self.readStatsBaseID if beenRead else self.statsBaseID
+        newID += str(index)
+        return newID
+
+    def _getSecondaryStatsID(self, index):
+        return self._getPrimaryStatsID(index, beenRead=True)
+
+    def _getStatsIDPair(self, index):
+        return self._getPrimaryStatsID(index), self._getSecondaryStatsID(index)
+
+    @staticmethod
+    def _newID(isFile=False, jobStoreID=None):
+        if isFile and jobStoreID:  # file associated with job
+            return jobStoreID+str(uuid.uuid4())
+        elif isFile:  # nonassociated file
+            return str(uuid.uuid4())
+        else:  # job id
+            return "job"+str(uuid.uuid4())
+
+    def _resolveEncryptionHeaders(self):
+        sseKeyPath = self.sseKeyPath
+        if sseKeyPath is None:
+            return {}
+        else:
+            with open(sseKeyPath) as f:
+                sseKey = f.read()
+            assert len(sseKey) == 32
+            encodedSseKey = base64.b64encode(sseKey)
+            encodedSseKeyMd5 = base64.b64encode(hashlib.sha256(sseKey).digest())
+            return {'x-goog-encryption-algorithm': 'AES256',
+                    'x-goog-encryption-key': encodedSseKey,
+                    'x-goog-encryption-key-sha256': encodedSseKeyMd5,
+                    "Cache-Control": "no-store"}
+
+    def _delete(self, jobStoreID, encrypt=True):
+        headers = self.encryptedHeaders if encrypt else self.headerValues
+        try:
+            key = self._getKey(jobStoreID, headers)
+        except NoSuchFileException:
+            pass
+        else:
+            try:
+                key.delete()
+            except boto.exception.GSResponseError as e:
+                if e.status == 404:
+                    pass
+        # best effort delete associated files
+        for fileID in self.files.list(prefix=jobStoreID):
+            try:
+                self.deleteFile(fileID)
+            except NoSuchFileException:
+                pass
+
+    def _getKey(self, jobStoreID=None, headers=None):
+        # gets remote key, in contrast to self._newKey
+        key = None
+        try:
+            key = self.files.get_key(jobStoreID, headers=headers)
+        except boto.exception.GSDataError:
+            pass  # see self._writeFile for more
+        if key is None:
+            raise NoSuchFileException(jobStoreID)
+        else:
+            return key
+
+    def _newKey(self, jobStoreID):
+        # does not create a key remotely. Careful- does not ensure key name is unique
```

Comment:
This is why I like comments to always start in upper-case. As soon as you have to sentences, it look awkward. Nit: `--` is commonly used in place of em dash.